### PR TITLE
[SPARK-42302][SQL] Assign name to _LEGACY_ERROR_TEMP_2135

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to assign name to _LEGACY_ERROR_TEMP_2135, "FAILED_PARSE_EMPTY_STRING".

### Why are the changes needed?

We should assign proper name to _LEGACY_ERROR_TEMP_*

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

`./build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite*"`

### CODE REVIEW

**Comment by itholic** [comment]
cc @srielau @MaxGekk @cloud-fan Please review this error class when you find some time

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -511,6 +511,12 @@
     ],
     "sqlState" : "38000"
   },
+  "FAILED_PARSE_EMPTY_STRING" : {
+    "message" : [
+      "Failed to parse an empty string for data type <dataType>."
+    ],
+    "sqlState" : "42P18"
```
Comment (srielau) [inline core/src/main/resources/error/error-classes.json:518]:
```suggestion
    "sqlState" : "42604"
```

We do know the data type. It's the literal we can't handle

**Review by srielau — APPROVED** [review]

File: core/src/main/resources/error/error-classes.json (hunk 1 above)
Comment (itholic) [reply core/src/main/resources/error/error-classes.json:518]: Thanks for the correction :-)

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala [hunk 2]
```diff
@@ -2608,11 +2609,12 @@ abstract class JsonSuite
...
-    val errMessage = intercept[SparkException] {
-      df.collect()
-    }.getMessage
-    assert(errMessage.contains(
-      s"Failed to parse an empty string for data type ${dataType.catalogString}"))
+    val e = intercept[SparkException] {df.collect()}
+    checkError(
+      exception = e.getCause.getCause.getCause.asInstanceOf[SparkRuntimeException],
+      errorClass = "FAILED_PARSE_EMPTY_STRING",
+      parameters = Map("dataType" -> s"${toSQLType(dataType)}")
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala:2616]: nit:
```suggestion
      parameters = Map("dataType" -> toSQLType(dataType))
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala [hunk 3]
```diff
@@ -1433,9 +1433,9 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
 
   def failToParseEmptyStringForDataTypeError(dataType: DataType): SparkRuntimeException = {
     new SparkRuntimeException(
-      errorClass = "_LEGACY_ERROR_TEMP_2135",
+      errorClass = "FAILED_PARSE_EMPTY_STRING",
       messageParameters = Map(
-        "dataType" -> dataType.catalogString))
+        "dataType" -> toSQLType(dataType)))
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:1438]:
```scala
      messageParameters = Map("dataType" -> toSQLType(dataType)))
```

File: core/src/main/resources/error/error-classes.json [hunk 4]
```diff
@@ -511,6 +511,12 @@
     ],
     "sqlState" : "38000"
   },
+  "FAILED_PARSE_EMPTY_STRING" : {
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:514]: This error class is pretty JSON specific. So, add JSON in the name. For example:
```suggestion
  "EMPTY_JSON_FIELD_VALUE" : {
```

File: core/src/main/resources/error/error-classes.json [hunk 5]
```diff
@@ -511,6 +511,12 @@
...
+    "message" : [
+      "Failed to parse an empty string for data type <dataType>."
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:516]: Please, tell about the SQL config `spark.sql.legacy.json.allowEmptyString.enabled` to users in the error message.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: core/src/main/resources/error/error-classes.json (hunk 5 above)
Comment (srielau) [reply core/src/main/resources/error/error-classes.json:516]: I'm conflicted about this. If something is LEGACY, then shouldn't it only apply to existing use cases.
Should we really advertise such configs?
On second thought, I'm not conflicted at all, FWIW I vote to NOT refer to legacy configs.

Comment (MaxGekk) [reply core/src/main/resources/error/error-classes.json:516]: ahh, I missed that it is a legacy config. I agree to don't advertise it.

**Comment by itholic** [comment]
Updated the comments, thanks!!

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master/3.4.
Thank you, @itholic and @srielau for review.

**Pull request closed without merging**